pro-planner-client/src/routes/LandingPage.jsx

```diff
@@ -1,146 +1,224 @@
-import React, { useState, useRef } from 'react';
+// url validation reference: https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url
+// url extraction reference: https://developer.mozilla.org/en-US/docs/Web/API/URL#Properties
+import React, {useState, useRef} from 'react';
 import {
     Button,
     Card,
     Col,
     Container,
     Form,
     Overlay,
-    OverlayTrigger,
     Row,
     Tooltip
 } from 'react-bootstrap';
-import { FiInfo } from 'react-icons/fi';
-import { useNavigate } from 'react-router-dom';
-import { useDispatch } from 'react-redux';
-
-import { changePlanType } from '../redux/planParamSlice';
-import { PLAN_TYPE } from '../constants';
+import {FiInfo} from 'react-icons/fi';
+import {useNavigate} from 'react-router-dom';
+import {useDispatch} from 'react-redux';
+import {changePlanType} from '../redux/planParamSlice';
+import {ERR_TYPE, PLAN_TYPE} from '../constants';
+import {setError} from "../redux/errorSlice";
+import axios from "axios";
+import {buildServerRoute} from "../helpers/Utils";
 
 const LandingPage = () => {
-    const [inputValue, setInputValue] = useState('');
-    const dispatch = useDispatch();
-    const navigate = useNavigate();
+        const [inputValue, setInputValue] = useState('');
+        const dispatch = useDispatch();
+        const navigate = useNavigate();
 
-    const tripButtonIcon = useRef(null);
-    const outingButtonIcon = useRef(null);
+        const tripButtonIcon = useRef(null);
+        const outingButtonIcon = useRef(null);
 
-    const [showTripTooltip, setShowTripTooltip] = useState(false);
-    const [showOutingTooltip, setShowOutingTooltip] = useState(false);
+        const [showTripTooltip, setShowTripTooltip] = useState(false);
+        const [showOutingTooltip, setShowOutingTooltip] = useState(false);
+        // may remove the localhost regex once deployed
+        const planUrlRegex = /^((https?:\/\/[^/]+\/[^/]+)|(https?:\/\/localhost:3000\/[^/]+))$/;
```

Comment:
might be a good idea to set the `localhost:3000` part to an env variable, so it'll be easy to dynamically change if we need.